### PR TITLE
Created Langfuse wrapper for langfuse integration

### PR DESCRIPTION
**Pull request by gabrielfior**
## Summary by CodeRabbit

- **New Features**
	- Integrated Langfuse support across the platform to enhance agent functionality with new language processing capabilities.
	- Added configuration settings for Langfuse API keys and host details.
	- Introduced a `LangfuseWrapper` class for creating Langfuse handlers with specific API keys and session ID.

### CODE REVIEW

File: prediction_market_agent_tooling/monitor/langfuse/langfuse_wrapper.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+from prediction_market_agent_tooling.config import APIKeys
+from prediction_market_agent_tooling.tools.utils import utcnow
+
+
+class LangfuseWrapper(BaseModel):
+    agent_name: str
+
+    @property
+    def session_id(self) -> str | None:
+        return f"{self.agent_name} - {utcnow()}"
```
Comment (kongzii) [inline prediction_market_agent_tooling/monitor/langfuse/langfuse_wrapper.py:13]: session_id will be different each time this property is accessed (even during he same session), is that desired?

File: prediction_market_agent_tooling/benchmark/agents.py [hunk 2]
```diff
@@ -18,6 +21,7 @@ def __init__(
         self.model = model
         self.agent_name = agent_name
         self.max_workers = max_workers  # Limit the number of workers that can run this worker in parallel threads
+        self.langfuse_wrapper = LangfuseWrapper(agent_name=self.__class__.__name__)
```
Comment (kongzii) [inline prediction_market_agent_tooling/benchmark/agents.py:24]: I think benchmark doesn't need langfuse, that's executed only locally. Do you see some usecase for it?

Comment (gabrielfior) [reply prediction_market_agent_tooling/benchmark/agents.py:24]: Depends on what we are trying to benchmark - if only performance, and that's captured in the Python reports, then happy to exclude this - if we are also keeping track of reasoning, etc, then I keep it.

@evangriffiths could you also weigh in here please?

File: prediction_market_agent_tooling/monitor/langfuse/langfuse_wrapper.py (hunk 1 above)
Comment (gabrielfior) [reply prediction_market_agent_tooling/monitor/langfuse/langfuse_wrapper.py:13]: The idea of `session_id` is to group traces together - my idea was to include here a kind of `run_identifier`, so every time we call `deploy_local` we will have an unique identifier `{agent}-{datetime_now}`.
This is then easily filterable, see below

image

Comment (kongzii) [reply prediction_market_agent_tooling/monitor/langfuse/langfuse_wrapper.py:13]: I get the idea that's cool. But shouldn't session_id be the same for the whole session?

Comment (gabrielfior) [reply prediction_market_agent_tooling/monitor/langfuse/langfuse_wrapper.py:13]: Refactored using @computed_property and @cache from [Pydantic docs](https://docs.pydantic.dev/2.7/api/fields/#pydantic.fields.computed_field)

File: prediction_market_agent_tooling/monitor/langfuse/langfuse_wrapper.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+from pydantic import BaseModel, computed_field
+
+from prediction_market_agent_tooling.config import APIKeys
+from prediction_market_agent_tooling.tools.utils import utcnow
+
+
+class LangfuseWrapper(BaseModel):
+    agent_name: str
+
+    @computed_field  # type: ignore
```
Comment (kongzii) [inline prediction_market_agent_tooling/monitor/langfuse/langfuse_wrapper.py:13]:
```suggestion
    @computed_field  # type: ignore[misc] # Mypy issue: https://github.com/python/mypy/issues/14461
```

**Review by kongzii — APPROVED** [review]

**Pull request merged**